airflow/dag_processing/manager.py

```diff
@@ -454,104 +437,66 @@ def deactivate_stale_dags(
-                        max_callbacks_per_loop, self.standalone_dag_processor, self.get_dag_directory()
-                    ):
-                        self._add_callback_to_queue(callback)
-                self._scan_stale_dags()
-                DagWarning.purge_inactive_dag_warnings()
-                refreshed_dag_dir = self._refresh_dag_dir()
-
-                if span.is_recording():
-                    span.add_event(name="_kill_timed_out_processors")
-                self._kill_timed_out_processors()
-
-                # Generate more file paths to process if we processed all the files already. Note for this
-                # to clear down, we must have cleared all files found from scanning the dags dir _and_ have
-                # cleared all files added as a result of callbacks
-                if not self._file_path_queue:
-                    self.emit_metrics()
-                    if span.is_recording():
-                        span.add_event(name="prepare_file_path_queue")
-                    self.prepare_file_path_queue()
+            loop_start_time = time.monotonic()
+            refreshed_dag_dir = self._refresh_dag_dir()
+
+            self._kill_timed_out_processors()
 
+            if not self._file_path_queue:
+                # Generate more file paths to process if we processed all the files already. Note for this to
+                # clear down, we must have cleared all files found from scanning the dags dir _and_ have
+                # cleared all files added as a result of callbacks
+                self.prepare_file_path_queue()
+                self.emit_metrics()
+            elif refreshed_dag_dir:
                 # if new files found in dag dir, add them
-                elif refreshed_dag_dir:
-                    if span.is_recording():
-                        span.add_event(name="add_new_file_path_to_queue")
-                    self.add_new_file_path_to_queue()
+                self.add_new_file_path_to_queue()
+
+            self._refresh_requested_filelocs()
+
+            self._start_new_processes()
 
-                self._refresh_requested_filelocs()
-                if span.is_recording():
-                    span.add_event(name="start_new_processes")
-                self.start_new_processes()
+            self._service_processor_sockets(timeout=poll_time)
 
-                # Update number of loop iteration.
-                self._num_run += 1
+            self._collect_results()
 
-                # Collect anything else that has finished, but don't kick off any more processors
-                if span.is_recording():
-                    span.add_event(name="collect_results")
-                self.collect_results()
+            if self.standalone_dag_processor:
+                for callback in self._fetch_callbacks():
+                    self._add_callback_to_queue(callback)
+            self._scan_stale_dags()
+            DagWarning.purge_inactive_dag_warnings()
```

Comment:
So in one of the next PRs, I assume we will need API endpoints for things like this